### ISSUE TITLE
Router.getChildContext throws away whatever was in the context which breaks tests that rely on both MemoryRouter as well as existing data in the context

### ISSUE THREAD

**Issue by jjinux** [ReactTraining/react-router#4970]
## Version
4.0.0

## Test Case
Sorry, none.

## Steps to reproduce
I'm writing a test using Jasmine, Enzyme, React Router, etc. I'm passing context to my component, which expects it. However, if I wrap my component with `<MemoryRouter>`, all of my sudden, my context data no longer shows up.

## Expected Behavior
`<Router>` should add router to the context without throwing away what was already in the context.

## Actual Behavior
`router` ends up in the context, and other things are thrown away.

I think the problem is in getChildContext in https://github.com/ReactTraining/react-router/blob/d601722f8515eea3910bb3a69313f3f2fc37df16/packages/react-router/modules/Router.js. It should call super, or something like that.

**Comment by pshrmn** [comment]
> Test Case
Sorry, none.

You're literally describing a failing test case. Can you please include that here?

**Comment by jjinux** [comment]
I wrote a test case:

[Verify_that_Router_getChildContext_doesn_t_wipe_the_existing_context.patch.txt](https://github.com/ReactTraining/react-router/files/921120/Verify_that_Router_getChildContext_doesn_t_wipe_the_existing_context.patch.txt)

I'm not bothering to create a PR because the test case passes. Hence, I have a suspicion that Enzyme's mount method doesn't pass the store object I have in context to MemoryRouter because MemoryRouter isn't asking for it. However, that means the components nested inside MemoryRouter don't get it either. If I don't wrap my component with MemoryRouter in my tests, but rather just put an instance of MemoryRouter into the context directly, the problem goes away. I also verified that if I create a wrapper component that explicitly puts the store into the context, and the wrapper component has MemoryRouter in it, and a real component in that, the problem doesn't happen.

I'm going to close this.
